mix.exs

```diff
@@ -49,7 +49,8 @@ defmodule Protein.Mixfile do
 
   defp deps do
     [
-      {:amqp, "~> 0.2"},
+      {:amqp, "~> 1.3.2"},
+      {:jsx, "~> 2.9", override: true},
```

Comment:
Libraries shouldn't ovverride deps (it's a very drastic measure that's a last resort for app code but I don't think it should be in a lib). Why do you do this? Isn't there an other way?

Reply:
There was a conflict in `jsx` versions, that's why this `override` is here.

Reply:
Can you explain the conflict? I.e. what conflicted with what? As I understand the only dependency in protein that pulls `jsx` is `amqp` so where does the conflict come from?

Reply:
```sh
➜  protein-elixir git:(feature/persistent-push-messages) ✗ mix deps.update jsx                                                                                                       <aws:shedul-developer>
Resolving Hex dependencies...

Failed to use "jsx" (version 2.10.0) because
  amqp (version 1.3.2) requires ~> 2.9
  rabbit_common (version 3.7.11) requires 2.9.0

Failed to use "jsx" (version 2.9.0) because
  amqp (version 1.3.2) requires ~> 2.9
  exjsx (version 4.0.0) requires ~> 2.8.0
  rabbit_common (version 3.7.11) requires 2.9.0

** (Mix) Hex dependency resolution failed, change the version requirements of your dependencies or unlock them (by using mix deps.update or mix deps.unlock). If you are unable to resolve the conflicts you can try overriding with {:dependency, "~> 1.0", override: true}
```

Reply:
I've solved this problem myself - by identifying `excoveralls` as the root cause of the conflict and removing this now-obsolete tool (but FYI it wasn't necessary to remove it - update also fixed the issue). I suggest next time you look for the root packages that start the dependency chain that clashes and loosening their version requirements / updating.

Reply:
> I suggest next time you look for the root packages that start the dependency chain that clashes and loosening their version requirements / updating.

Within' `mix.lock` I assume?